### PR TITLE
[FLINK-36007][[cdc-composer] cdc load factory and add jar in once search

### PR DESCRIPTION
**Pull request by loserwang1024**
As shown https://issues.apache.org/jira/browse/FLINK-36007 , once search is enough for cdc load factory and add jar url.

### CODE REVIEW

**Comment by loserwang1024** [comment]
@leonardBang @ruanhang1993 ,CC

File: flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/utils/FactoryDiscoveryUtils.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import org.apache.flink.cdc.common.annotation.Internal;
 import org.apache.flink.cdc.common.factories.Factory;
 
+import org.codehaus.commons.nullanalysis.NotNull;
```
Comment (leonardBang) [inline flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/utils/FactoryDiscoveryUtils.java:23]: import javax.annotation.Nonnull;

**Review by leonardBang — COMMENTED** [review]
Thanks @loserwang1024 for the optimization, left one minor comment

File: flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/utils/FactoryDiscoveryUtils.java (hunk 1 above)
Comment (loserwang1024) [reply flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/utils/FactoryDiscoveryUtils.java]: done

**Review by leonardBang — APPROVED** [review]

**Pull request merged**